### DISCUSSION TITLE
Multiclass classification on Arabic Tweets

### DISCUSSION THREAD

**Discussion by fg214** [aub-mind/arabert#126] [Q&A]
Hi Wissam!

Im an absolute beginner to BERT, and would really appreciate your insight. I'm collected Arabic hate-speech (misogynistic) Tweets, and I want to use AraBERT to build a multiclass classifier to distinguish the different contexts the tweets occur in (4 classes). 

Which model/library do you think is best for that? Would Fast-Bert work? 

Best,
Farah

**Comment by WissamAntoun** [comment]
I would recommend checking the example notebook here https://colab.research.google.com/drive/1P9iQHtUH5KUbTVtp8B4-AopZzEEPE0lw?usp=sharing . This can handle multi-class and multi-label. You just need to change the problem_type parameter in the config when instantiating the model using .from_pretrained() (https://github.com/huggingface/transformers/blob/master/src/transformers/models/bert/modeling_bert.py#L1561)

**Reply by ReemAlJunaid** [reply]
I'm following your code to perform text classification for set of questions, after training and saving the model, I want to see the predictions for some texts. I wrote the following code:

`from transformers import pipeline,AutoModel, AutoModelForSequenceClassification, AutoTokenizer

model_name = 'aubmindlab/bert-base-arabertv02'

arabert_model = AutoModelForSequenceClassification.from_pretrained('/gdrive/MyDrive/LabelModel')

tokenizer = AutoTokenizer.from_pretrained(model_name)

text = "أين وقعت غزوة بدر؟"
#{'كيان': 0, 'تقريري': 1, 'حدث': 2, 'رقم': 3, 'عاقل': 4, 'موقع': 5, 'وصف': 6}
pipe = pipeline(model=arabert_model, tokenizer=tokenizer)
pipe(text)  `

First, it gave me this error: 
`OSError: Error no file named pytorch_model.bin, tf_model.h5, model.ckpt.index or flax_model.msgpack found in directory /gdrive/MyDrive/LabelModel.`

Then I solved this problem by renaming the training_args.bin file to  pytorch_model.bin as in [this link ](https://stackoverflow.com/questions/62961627/oserror-error-no-file-named-pytorch-model-bin-tf-model-h5-model-ckpt-in)

After that, another error exists:
`AttributeError: 'TrainingArguments' object has no attribute 'keys'`

How can I solve this?

**Reply by aalmhirat** [reply]
Hello Mr. Wissam
I hope to be fine 
 
My case is : 
I'm collected dataset about racism in Arabic social media ( Twitter/Facebook) and classify to (4) class (Race, Religion, Disabilities, Other), my dataset is ready as csv
I supervised for labelling dataset to classify racism to multiclass,
I'm beginner in  AraBert, I need help how to using AraBert  can you guide me how learning steps for that .
Thanks
Best regards
Ashraf almhirat
Jordan

**Comment by aalmhirat** [comment]
> Hi Wissam!
> 
> Im an absolute beginner to BERT, and would really appreciate your insight. I'm collected Arabic hate-speech (misogynistic) Tweets, and I want to use AraBERT to build a multiclass classifier to distinguish the different contexts the tweets occur in (4 classes).
> 
> Which model/library do you think is best for that? Would Fast-Bert work?
> 
> Best, Farah

hi Farah also my case is the same for you but another content I'm collected dataset about racism and classification for multiclass (5)
i need help for using AraBert to do that and I'm beginner how can learning steps . thanks for you
